### ISSUE TITLE
rdflib does not support upper case SERVICE

### ISSUE THREAD

**Issue by zwelz3** [RDFLib/rdflib#1242]
With the latest updates to `service` queries in rdflib 5.0 (60ad0aa), there is still no support for upper case `SERVICE` calls in rdflib, and no library indication that the syntax is invalid when upper case vs lower case. 

I was unable to locate a ticket to address this behavior. Is this desired? Should users be required to use lower case for `service` where all other SPARQL keywords allow either?

Minimal example:
```python
import rdflib

res = rdflib.graph.Graph().query(
    """\
    PREFIX wd: <http://www.wikidata.org/entity/>
    PREFIX wdt: <http://www.wikidata.org/prop/direct/>
    SELECT ?item
    WHERE {
        SERVICE <https://query.wikidata.org/sparql>
        {
            SELECT ?item
            WHERE {
                ?item wdt:P31 wd:Q146 .
            }
            LIMIT 5
        }
    }
    """
)
assert len(res) > 0
```

The above query only works with lower case `service`.

**Comment by white-gecko** [comment]
You are completely right, the SERVICE keyword shoukd also be accepted in upper case.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by rchateauneu** [comment]
Maybe it is here ?
In rdflib.plugins.sparql.evaluate

def evalServiceQuery(ctx, part):
    res = {}
    match = re.match('^service <(.*)>[ \n]*{(.*)}[ \n]*$',
                     part.get('service_string', ''), re.DOTALL)

Possibly replace the regex  flag should be: re.DOTALL | re.IGNORECASE

**Comment by rchateauneu** [comment]
It is fixed in master:
    match = re.match(
        "^service <(.*)>[ \n]*{(.*)}[ \n]*$",
        part.get("service_string", ""),
        re.DOTALL | re.I,
    )

**Comment by white-gecko** [comment]
So it was fixed by #1041 already. It will be part of the next release. If you need the fix right now, you can use the master branch but be aware, it is under development.
